### PR TITLE
os/bluestore: unify throttling model

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by ivancich** [comment]
This looks good.

So cost per io is in units of bytes, in other words, how many bytes could have been written during an average prep/seek time?

**Comment by liewegas** [comment]
something like that!  it's not really that precise (some seeks go away 
because we have sequential ios, etc) but the idea is to smoosh it all into 
one type of units.

**Comment by markhpc** [comment]
In tests last night this seemed to help on spinning disk, but I'm not sure it's helping (and might be hurting) on HDD/NVMe and NVMe only tests.  Granted, these are only small preliminary tests on a single OSD.
[preextend_tests.xlsx](https://github.com/ceph/ceph/files/896468/preextend_tests.xlsx)

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -7786,12 +7784,19 @@ void BlueStore::_kv_sync_thread()
 	  }
 	}
       }
-      if (num_aios) {
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.cc]: why is this check removed?

Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: it was an optimization of dubious value (avoid iterating over the list if we know there are no aios), but with the throttle change we have to do the iteration anyway.

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Pull request merged**